apps/commerce-api/src/main/kotlin/com/loopers/interfaces/event/OrderEventListener.kt

```
@@ -0,0 +1,157 @@
+     * - ObjectOptimisticLockingFailureException 발생 시 쿠폰 중복 사용으로 간주
+     */
+    @TransactionalEventListener(phase = TransactionPhase.BEFORE_COMMIT)
+    fun handleOrderCreatedForCoupon(event: OrderEvent.OrderCreated) {
+        log.info("주문 생성 - 쿠폰 사용 처리 시작: orderId=${event.orderId}")
+
+        // 할인이 있는 경우에만 쿠폰 사용 처리
+        event.couponId?.let { couponId ->
+            try {
+                couponService.applyCoupon(event.userId, event.couponId)
+                log.info("쿠폰 사용 완료: orderId=${event.orderId}, userId=${event.userId}, couponId=${event.couponId}")
+            } catch (e: ObjectOptimisticLockingFailureException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)